datasets/flwr_datasets/partitioner/iid_partitioner_test.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2023 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Partitioner tests."""
+
+
+import unittest
+
+from datasets import Dataset
+from flwr_datasets.partitioner.iid_partitioner import IidPartitioner
+
+
+class TestIidPartitioner(unittest.TestCase):
+    """Test IidPartitioner."""
+
+    def setUp(self) -> None:
+        """Create a dummy dataset with 100 rows, numerical features, and labels."""
+        data = {
+            "features": list(range(100)),
+            "labels": [i % 2 for i in range(100)],
+        }
+        self.dataset = Dataset.from_dict(data)
+        self.num_partitions = 10
+        self.partition_size = self.dataset.num_rows // self.num_partitions
+        self.sampler = IidPartitioner(num_partitions=self.num_partitions)
+        self.sampler.dataset = self.dataset
```

Comment:
I'd suggest to
1. Move this from `setUp` to the actual test method
2. Merge the two test methods below into a new one (`test_load_partition_10`) with two asserts at the end
3. Create additional cases, especially covering edge cases. Values for `num_partitions` that are expected to work: 1, 100. Values that are not expected to work, and should result in a well-defined error: 0, 101.

Reply:
Re1: I created a utility funtion not method to accomplish the simillar thing.
Re2: I think that two asserts in a single test is a bad coding practice (violates signle responsibilty principle but it's also simply hard to see what fails in an easy manner). I made it work with single asserts
Re3: I addeed edge cases. I parameterized the all the paritioning tests.